src/drivers/drv_spi_a7105.c

```diff
@@ -0,0 +1,216 @@
+#include "drv_spi_a7105.h"
+#include "drv_spi.h"
+#include "drv_time.h"
+#include "drv_exti.h"
+#include "project.h"
+
+#if defined(USE_A7105)
+
+static spi_bus_device_t bus = {
+    .port = A7105_SPI_PORT,
+    .nss = A7105_NSS_PIN,
+
+    .auto_continue = true,
+};
+
+//------------------------------------------------------------------------------
+static volatile uint32_t irq_timestamp = 0;
+static volatile bool irq_triggered = false;
+
+//------------------------------------------------------------------------------
+// See: drv_exti.c
+// This is called when the A7105 GPIO1 pin goes high triggering our external interrupt
+void a7105_handle_busy_exti(bool pin_state) {
+  if (pin_state) {
+    irq_timestamp = time_micros();
+    irq_triggered = true;
+  }
+}
+
+//------------------------------------------------------------------------------
+void a7105_write_reg(a7105_reg_t reg, uint8_t data) {
+  spi_txn_t *txn = spi_txn_init(&bus, NULL);
+  spi_txn_add_seg_const(txn, reg);
+  spi_txn_add_seg_const(txn, data);
+  spi_txn_submit_continue(&bus, txn);
+}
+
+static void a7105_write_multi(uint8_t reg, const uint8_t *data, uint8_t len) {
+  spi_txn_t *txn = spi_txn_init(&bus, NULL);
+  spi_txn_add_seg_const(txn, reg);
+  spi_txn_add_seg(txn, NULL, data, len);
+  spi_txn_submit(txn);
+
+  // needs to sync ???
+  spi_txn_continue_ex(&bus, true);
```

Comment:
Using `true` or `false` with some very limited testing doesn't seem to affect the behavior. Since I've been using it like this for a while I'd like to retain this code as it is.

*But* I would like to review the entire SPI implementation a bit more as I used the FrSky (CC2500) code as the basis for this source file rather than taking a deep dive into understanding how the SPI implementation works. It seems it can be blocking or not depending on how you call into it.

In my test project (outside of Quicksilver) I use a soft spi approach (bit bang) and it works great. My link quality for AFHDS is rock solid (666 packets per second). But in Quicksilver it fluctuates every now and then and the only real differences between my test project and Quicksilver is possibly the looptime (I know my test project is rock solid 250us) and the soft spi (which is synchronous).

tl;dr I want to revisit this SPI implementation with your help but hope we can accept this for now

Reply:
yeah, i would bet the fluctuations are down to a loop taking longer and it not returning to rx code in time. We got a lot of other stuff going on at the same time. i have been working on a scheduler https://github.com/BossHobby/QUICKSILVER/tree/feature-scheduler to ensure crictical paths, like the rx code, are always ran on time.